ludwig/api.py

```diff
@@ -665,6 +665,7 @@ def on_epoch_end(self, trainer, progress_tracker, save_path):
                             else:
                                 calibrator.train_calibration(validation_set, VALIDATION)
                         if not skip_save_model:
+                            self.model = trainer.model
```

Comment:
Ensure that any changes to the model object held by the trainer class are reflected in the model here.

Reply:
Should this just be a return type of the trainer.train method? Otherwise it seems we're relying on some inherent statefulnes that shouldn't be guaranteed.

Reply:
That's reasonable– I was basing it off of the Calibrator logic, which uses `trainer.model` directly for some reason: https://github.com/ludwig-ai/ludwig/blob/0fdcb98bcd926c7ad83974743cc21f0953261986/ludwig/api.py#L646

Happy to change this line of code as well if we think it is also a code smell.

Reply:
Alright, updated both this line and the Calibrator line to use `self.model`.